rmgpy/data/thermo.py

```diff
@@ -421,6 +421,66 @@ def processOldLibraryEntry(self, data):
         """
         return processOldLibraryEntry(data)
 
+    def copyData(self, source, destination):
+        """
+        This method copys the ThermoData object and all meta data
+        from source to destination
+        Args:
+            source: The entry for which data is being copied
+            destination: The entry for which data is being overwritten
+
+        """
+        destination.data = source.data
+        destination.reference = source.reference
+        destination._longDesc = source._longDesc
+        destination._shortDesc = source._shortDesc
+        destination._longDesc = source.longDesc
+        destination._shortDesc = source.shortDesc
+        destination.rank = source.rank
+        destination.referenceType = source.referenceType    
+
+
+    def removeGroup(self, groupToRemove):
+        """
+        Removes an entry from the database. For thermo groups we also,
+        need to re-point any unicode thermoData that may have pointed
+        to the entry.
+        """
+
+        #First call base class method
+        Database.removeGroup(self, groupToRemove)
+
+        parentR = groupToRemove.parent
+
+        #If the parent.data points toward the entryToRemove, then copy
+        #the data and all meta-data
+        if parentR.data == groupToRemove.label:
+            parentR.data = deepcopy(groupToRemove.data)
+            #if actually a ThermoGroup, then copy metaData
+            if not isinstance(parentR.data, basestring):
+                parentR.reference = groupToRemove.reference
+                parentR._longDesc = groupToRemove._longDesc
+                parentR._shortDesc = groupToRemove._shortDesc
+                parentR._longDesc = groupToRemove.longDesc
+                parentR._shortDesc = groupToRemove.shortDesc
+                parentR.rank = groupToRemove.rank
+                parentR.referenceType = groupToRemove.referenceType
+
+        #look for other pointers that point toward entry
+        for entryName, entry in self.entries.iteritems():
```

Comment:
I think this `for` should be combined with the `if` statement starting at line 457 in order to make it more concise.